subsys/net/ip/net_shell.c

```diff
@@ -2659,24 +2659,38 @@ static enum net_verdict _handle_ipv6_echo_reply(struct net_pkt *pkt,
 						struct net_ipv6_hdr *ip_hdr,
 						struct net_icmp_hdr *icmp_hdr)
 {
-	ARG_UNUSED(icmp_hdr);
+	struct net_icmpv6_echo_req *echo_repl = (struct net_icmpv6_echo_req *)(icmp_hdr + 1);
```

Comment:
not good, see my comment in ipv4_echo_reply cb

Reply:
Yea I didn't like that either.
So I understood you right, I'll change it so that the handler gets the `struct net_icmpv4_echo_req` instead of `struct net_icmp_hdr` .

Reply:
nvm, I figured `NET_PKT_DATA_ACCESS_CONTIGUOUS_DEFINE` / `net_pkt_get_data_new()` is the way to go.